app.py

```diff
@@ -59,6 +59,26 @@ def node_register():
     return jsonify({})
 
 
+@app.route('/v1/proposals', methods=['GET'])
+def proposals():
+    node_key = request.args.get('node_key')
+    node = Node.query.get(node_key)
+    if not node:
+        return jsonify(error='node by node_key not found'), 400
+
+    try:
+        config = json.loads(node.connection_config)
+    except ValueError:
+        return jsonify(error='cannot deserialize node configuration'), 400
+
+    service_proposal = config.get('service_proposal')
+    if service_proposal is None:
+        return jsonify(error='node config does not contain service proposal'), 400
+
+    return jsonify({'proposals': [service_proposal]})
```

Comment:
Dont need nesting with 'proposals' key

Reply:
Returning array as response have disadvantages:
- If you want to add some extra data, you can't
- In case of error, dictionary of `{"errors": "message"}` is returned, so returning dictionary in some cases and `array` in other cases is not very consistent.

Reply:
Absolutely agreed

Reply:
I agree on nesting - list as root response element is not extendable without breaking existing api consumers, as you mentioned i.e adding extra data - a good example: adding paging support for proposal list later.

Reply:
Ok, convinced.
But we do arrays in other places

Reply:
Not on my watch :) it should be fixed too.